apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClientBuilder.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.Grpc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.PubSub.V1;
+
+// TODO: Should the TopicName just be a regular property?
```

Comment:
Yes, and we can validate that it's present on build same as we do with the rest. Granted the rest has defaults, but still, this is a builder.

Reply:
Mmm... it just feels odd for something that's required not to be specified in the constructor. I'm trying to think of similar situations we've got in other libraries - will check Storage and BigQuery.

Reply:
`BigQueryClientBuilder` has it as a normal property, validated - so yes, I'll switch to that pattern for consistency.

Reply:
Maybe we should introduce new client static Create/CreateAsync overloads without any settings for "I just want to create it for this topic, using defaults". Let's talk about this more in our meeting :)